### ISSUE TITLE
Type check `name` based on `formValidations`?

### ISSUE THREAD

**Issue by kentcdodds** [brophdawg11/remix-validity-state#9]
It would be awesome if this typechecked:

```
const confirmPasswordField = useValidatedInput({
  name: 'confirm-password',
  formValidations,
  errorMessages,
  serverFormInfo: actionData?.serverFormInfo,
})
```

So if my `formValidations` didn't have a `confirm-password` field I'd get a warning on the `name` (or vice versa, I'd just like to know I made a typo).

**Comment by brophdawg11** [comment]
Time to dust off my TS chops!  Should be possible with some `keyof` checks I think

**Comment by brophdawg11** [comment]
Should be available in `0.5.0`

**Comment by kentcdodds** [comment]
Thank you!
